test/utils.jl

```diff
@@ -1,6 +1,65 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 import Pkg
 
+const LOADED_DEPOT = joinpath(@__DIR__, "loaded_depot")
+const FORCE_CLEAN_UNITS = false
+
+function test_depot_setup(; loaded_depot=false)
+    if FORCE_CLEAN_UNITS
+        return mktempdir()
+    end
+
+    Pkg.UPDATED_REGISTRY_THIS_SESSION[] = loaded_depot
```

Comment:
This still doesn't prevent the registry from updating on `pkg> update`, which we do quite often. Maybe we should add https://github.com/JuliaLang/Pkg.jl/pull/1315 after all.